index.js

```diff
@@ -7,33 +27,221 @@ function ask(questionText) {
-  if (validTransitions.includes(newState)) {
-    currentState = newState;
-  } else {
-    throw 'Invalid state transition attempted - from ' + currentState + ' to ' + newState;
+async function moveToRoom(churchSt2) {
+  churchSt2.message();
+  let cursor = await ask (">_")
+  if (cursor.toUpperCase === 'Walk' || 'Go' || 'Leave') {
+    moveToRoom(midTown);
+  }
+  else {
+    moveToRoom(midTown);
   }
 }
 
-start();
+async function moveToRoom(midTown) {
+  midTown.message();
+  let entry = await ask ("Try a simple 5-digit code, such as 12345")
+  if (entry === '12345') {
+    console.log("You're in! Enter the foyer and read the sign.")
+    moveToRoom(acadFoyer);
+  }
+  else {
+    console.log("Sorry, wrong code. Try again.")
+    moveToRoom(midTown);
+  }
+}
 
-async function start() {
-  const welcomeMessage = `182 Main St.
-You are standing on Main Street between Church and South Winooski.
-There is a door here. A keypad sits on the handle.
-On the door is a handwritten sign.`;
-  let answer = await ask(welcomeMessage);
-  console.log('Now write your code to make this work!');
-  process.exit();
+async function moveToRoom(acadFoyer) {
+  acadFoyer.message();
+  let cursor = await ask (">_")
+  if (cursor.toUpperCase === 'Take' || 'Take Sign' || 'Steal' || 'Steal Sign') {
+    console.log("You can't take the sign. Stealing isn't cool, man.")
+  }
+  else if (cursor.toUpperCase === 'Go' || 'Go Upstairs' || 'Up' || 'Upstairs') {
+    moveToRoom(bcaLab);
+  }
 }
+
+async function moveToRoom(bcaLab) {
+  bcaLab.message();
+  let cursor = await ask (">_")
+  if (cursor.toUpperCase === 'Copy' || 'Paper') {
+    roomInv.pop();
+    playerInv.push("7 Days paper")
+  }
+  else if (cursor.toUpperCase === 'Coffee') {
+    roomInv.pop();
+    playerInv.push("Coffee")
+  }
+  else if (cursor.toUpperCase === 'Tea') {
+    roomInv.pop();
+    playerInv.push("Tea")
+  }
+  else if (cursor.toUpperCase === 'Go' || 'Leave' || 'Down' || 'Downstairs') {
+    console.log("That's enough lecture for today. State machines are really confusing!")
+    moveToRoom(midTown2);
+  }
+}
+
+async function moveToRoom(midTown2) {
+  midTown2.message();
+  let cursor = await ask (">_")
+  if (cursor.toUpperCase === 'Walk' || 'Walk Downtown' || 'Go' || 'Go Downtown') {
+    moveToRoom(churchStreet3);
+  }
+  else {
+    console.log("What do you feel like doing?")
+    moveToRoom(midTown2);
+  }
+}
+
+async function moveToRoom(churchStreet3) {
+  churchStreet3.message();
+  let cursor = await ask (">_")
+  if (cursor.toUpperCase === 'Walk') {
+    moveToRoom(apartMent);
+  }
+  else if (cursor.toUpperCase === 'Ride') {
+    playerInv.pop();
+    moveToRoom(apartMent);
+  }
+}
+
+async function moveToRoom(apartMent) {
+  apartMent.message();
+  let cursor = await ask (">_")
+  if (cursor.toUpperCase === 'Drop') {
+    playerInv.length = 0
+  }
+  else if (cursor.toUpperCase === 'Done') {
+    console.log("\n>_")
+    process.exit();
+  }
+}
```

Comment:
Delete all of these. You should only have one `moveToRoom` function which should be able to take **any** room as an argument and move you into that room.  Remember when you are defining a function the argument variable creates a **new scoped** variable with that name, it doesn't point to the variable of the same name defined above. `apartMent` here does not point to your `apartMent` Room.